### PR TITLE
[WIP] Pull-to-refresh

### PR DESCRIPTION
**Pull request by Huddie**
This is a WIP the original concept https://github.com/GitHawkApp/GitHawk/issues/1580#issuecomment-410926148
had the branches in there but it seemed a bit cluttered.

 Thoughts and feedback are appreciated!

### CODE REVIEW

**Comment by Huddie** [comment]
I forgot again! Whoops. I’ll update that. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Huddie** [comment]
TODO

- [ ] Switch to snapkit

File: Classes/Views/GitHawkRefreshControl.swift [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    private func updateProgress(_ yOffset: CGFloat) {
+        // Convert to 0 - 1 scale
+        progress = (yOffset * 100 / startingHeight) / 100
+    }
+    
+    private func setPlacement(percent: CGFloat) {
+        let adjPercent = percent < 1
+            ? -startingHeight + (percent * startingHeight)
+            : 1
+        contentView.transform = CGAffineTransform(
```
Comment (jdisho) [inline Classes/Views/GitHawkRefreshControl.swift]: add new line

File: Classes/Views/GitHawkRefreshControl.swift [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            translationX: 1,
+            y: adjPercent
+        )
+    }
+    
+    private func scale(percent: CGFloat) {
+        let adjPercent = percent < 1
+            ? percent
+            : 1
+        self.transform = CGAffineTransform(
```
Comment (jdisho) [inline Classes/Views/GitHawkRefreshControl.swift]: add new line

**Comment by rnystrom** [comment]
Closing as its stale

**Pull request closed without merging**